signer/core/cliui.go

```diff
@@ -242,6 +244,24 @@ func (ui *CommandlineUI) OnApprovedTx(tx ethapi.SignTransactionResult) {
 }
 
 func (ui *CommandlineUI) OnSignerStartup(info StartupInfo) {
+	accounts, err := ui.api.ListAccounts(context.Background())
+	if err != nil {
+		fmt.Print("error listing accounts", err)
+	}
+	if len(accounts) == 0 {
+		fmt.Println("No accounts known to Clef")
+	} else {
+		// account info to string for nicer printing
+		var addresses string = "\n"
+		fmt.Println("Accounts known to Clef:")
+		for i, account := range accounts {
+			// concat string to avoid repeating "INFO" on terminal
+			addresses += fmt.Sprintf("Account %v: %s at %s", i, account.Address, account.URL)
+			addresses += "\n"
+		}
+		fmt.Print(addresses)
+		fmt.Println()
+	}
```

Comment:
Generally correct, but two nitpicks. 

1. We are in an rpc callback, a remote process has issued a remote procedure call. Generally, it might be waiting for a response (but in this particular case, `OnSignerStartup` is an answer-less request. Therefore, I think it's a good idea to end this call as soon as possible, and return to the caller and let the caller continue -- instead of _inside the handler_ do a call back for listing accounts. 

So, the TLDR; use this method for bootstrapping only, do the account-listing asynchronously by placing this clause inside a  go func(){ ... }()


2. When printing to `os.Stdout`, there are other things that can happen. There may be `Log` events happening, and there may be incoming requests prompting the user for action, in different threads/routines. 

So it is better to do do the printing only once, to ensure that the chunk of text that we want to display is displayed in one piece, not broken up by other messages/prompts. That can be done either by souping up the lines to write into a `[]string` and then using `strings.Join`, or by using a `strings.Builder`.

This is something that I have _not_ done a very good job of myself, in the rest of this file.

Reply:
Actually, ignore the second point. Let's just leave it be, and maybe we can fix all of those at the same time, in a separate PR